src/repr/scalar/mod.rs

```diff
@@ -745,6 +745,46 @@ impl Default for Interval {
     }
 }
 
+impl std::ops::Add for Interval {
+    type Output = Self;
+    // We need subtraction and boolean operators to sensibly
+    // add intervals to one another.
+    #[allow(clippy::suspicious_arithmetic_impl)]
+    fn add(self, other: Self) -> Self {
+        let (is_dur_sum_pos, dur_sum) = if self.is_positive_dur == other.is_positive_dur {
+            (self.is_positive_dur, self.duration + other.duration)
+        } else if self.duration > other.duration {
+            (self.is_positive_dur, self.duration - other.duration)
+        } else {
+            (other.is_positive_dur, other.duration - self.duration)
+        };
+
+        Self {
+            months: self.months + other.months,
+            duration: dur_sum,
+            is_positive_dur: is_dur_sum_pos,
```

Comment:
fwiw if you did `let (is_positive_dur, duration)` this could be changed:

```suggestion
            duration,
            is_positive_dur,
```

But I suppose you might have chosen these variables intentionally to make the logic more clear?